aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/service/AmplifyTransferService.kt

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3.service
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.Service
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.ServiceConnection
+import android.net.ConnectivityManager
+import android.os.Binder
+import android.os.Build
+import android.os.Handler
+import android.os.IBinder
+import android.os.Looper
+import androidx.annotation.RequiresApi
+import androidx.annotation.VisibleForTesting
+import androidx.core.app.NotificationCompat
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferNetworkLossHandler
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferStatusUpdaterAccessor
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.storage.s3.R
+
+internal class AmplifyTransferService : Service() {
+
+    private val log = Amplify.Logging.forNamespace("amplify:aws-s3")
+
+    private val binder = LocalBinder()
+
+    /**
+     * registers a BroadcastReceiver to receive network status change events. It
+     * will update transfer records in database directly.
+     */
+    private var transferNetworkLossHandler: TransferNetworkLossHandler? = null
+
+    /**
+     * A flag indicates whether or not the receiver has is started the first time.
+     */
+    private var isReceiverNotRegistered = true
```

Comment:
I think it would be clearer to change this variable to isReceiverRegistered, so it's a bit easier to understand what !isReceiverNotRegistered means.

Reply:
agree. changed. Some of this I had moved from TransferService and tried to keep as close in appearance as possible, but makes sense to make improvements like this.